File: 92.反转链表-ii.js
/*
 * @lc app=leetcode.cn id=92 lang=javascript
 *
 * [92] 反转链表 II
 */
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} m
 * @param {number} n
 * @return {ListNode}
 */
var reverseBetween = function(head, m, n) {
    let counter = 1
    let dummy = new ListNode()
    dummy.next = head
    let traNode = head
    let preNode = dummy
    let preNodeOfM = null
    while(traNode){
        let now = traNode
        let pre = preNode
        let next = traNode.next
        traNode = traNode.next
        if(counter===m){
            preNodeOfM = pre
        }else if(counter>m){
            now.next = pre
        }else if(counter === n ){
            now.next = pre
            preNodeOfM.next.next = next
            preNodeOfM.next = now
            break
        }
        counter++
        preNode = preNode.next

    }

    return dummy.next
};

File: 92.反转链表-ii.java
/*
 * @lc app=leetcode.cn id=92 lang=java
 *
 * [92] 反转链表 II
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseBetween(ListNode head, int m, int n) {
        ListNode dummy = ListNode();
        ListNode before = dummy;
        ListNode after = ListNode();
        ListNode node = head;
        dummy.next = head;
        int i = 1;
        while(node!=null){
            if(i<m){
                before = node;
            }else if(i==m){

            }else if(i>n){
                break;
            }else{
                before = node;
            }
            node = node.next;
        }

        return dummy.next;
    }
}
// @lc code=end

